crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/service/RepOrderService.java

```diff
@@ -122,4 +129,95 @@ public ReviewResult getReviewResult(ApiIOJAppeal apiIOJAppeal) {
         return null;
     }
 
+    public ApiCrownCourtSummary determineCrownRepType(CrownCourtsActionsRequestDTO requestDTO) {
+        ApiCrownCourtSummary crownCourtSummary = requestDTO.getCrownCourtSummary();
+        /*
+            PROCEDURE determine_crown_rep_type (p_app_obj    IN OUT    application_type) IS BEGIN -- determine_crown_rep_type
+       if p_app_obj.crown_court_overview_object.crown_court_summary_object.CC_reporder_decision is not null
+       THEN
+       */
+        if (crownCourtSummary.getRepOrderDecision() != null) {
+            /*
+              IF p_app_obj.mags_outcome_object.outcome IN ('SENT FOR TRIAL','COMMITTED FOR TRIAL')
+              THEN
+            */
+            if (requestDTO.getMagCourtOutcome() == MagCourtOutcome.SENT_FOR_TRIAL
+                    || requestDTO.getMagCourtOutcome() == MagCourtOutcome.COMMITTED_FOR_TRIAL) {
+                determineRepTypeByDecisionReason(requestDTO, crownCourtSummary);
+            }
+            determineRepTypeByRepOrderDecision(requestDTO, crownCourtSummary);
```

Comment:
Should this and the following conditions not be in `else if` or an `else` block? Doesn't make sense to run the code if we don't need to.

Reply:
Leaving this as it is to be inline with the existing stored procedure logic.